pints/_log_pdfs.py

```diff
@@ -115,6 +115,161 @@ def sample(self, n=1):
         raise NotImplementedError
 
 
+class PooledLogPDF(LogPDF):
+    r"""
+    Combines :math:`m` :class:`LogPDFs<pints.LogPDF>`, each with :math:`n`
+    parameters, into a single LogPDF where :math:`k` parameters are "pooled"
+    (i.e. have the same value for each LogPDF), so that the resulting combined
+    LogPDF has :math:`m (n - k) + k` parameters.
+
+    This is useful for e.g. modelling the time-series of multiple individuals
+    (each individual defines a separate :class:`LogPDF`), and some parameters
+    are expected to be the same across individuals (for example, the noise
+    parameter across different individuals within the same experiment).
```

Comment:
Can we get some textbook reference for this in general?

Reply:
So that users can learn more about this if they are unsure whether they are using the right thing/approach for their problem!

Reply:
I am not sure whether there is a text book reference. @ben18785 Would you know of one?

Reply:
OK, if there's no "textbook reference", I remember you mentioned this has been used in PKPD modelling, if there's a ref or method paper that explains this thoroughly, maybe worth pointing to?

Reply:
It is used in the PKPD modelling community, but the paper I've seen so far tend to be not very technical. I.e. they use it without reasoning why they use it. But I hope to include something on this in a future paper. :D

Reply:
If it is not used so often, then it could even be more useful to have example references here. Such that users like me who can't come up with practical examples as to when to use this method can have something to refer to.